docs/further_useful_features.rst

```diff
@@ -0,0 +1,459 @@
+
+    dm = store_dataframes_as_dataset(
+        store_factory, "partitioned_dataset", df, partition_on="E"
+    )
+
+
+Of interest here is ``dm.partitions``:
+
+.. ipython:: python
+
+    sorted(dm.partitions.keys())
+
+
+We can see that partitions have been stored in a way which indicates the
+specific value for the column on which partitioning has been performed.
+
+Partitioning can also be performed on multiple columns; in this case, columns
+should be specified as a list:
+
+.. ipython:: python
+
+    duplicate_df = df.copy()
+    duplicate_df.F = "bar"
+
+    dm = store_dataframes_as_dataset(
+        store_factory,
+        "another_partitioned_dataset",
+        [df, duplicate_df],
+        partition_on=["E", "F"],
+    )
+    sorted(dm.partitions.keys())
+
+
+Note that, since 2 dataframes have been provided as input to the function, there are
+4 different files created, even though only 2 different combinations of values of E and
+F are found, ``E=test/F=foo`` and ``E=train/F=foo`` (However, these 4 physical partitions
+can be read as just the 2 logical partitions by using the argument
+``concat_partitions_on_primary_index=True`` at reading time).
+
+For datasets consisting of multiple tables, explicit partitioning on columns can only be
+performed if the column exists in both tables and is of the same data type: guaranteeing
+that their types are the same is part of schema validation in ``kartothek``.
+
+For example:
+
+.. ipython:: python
+
+    df.dtypes
+    different_df = pd.DataFrame(
+        {"B": pd.to_datetime(["20130102", "20190101"]), "L": [1, 4], "Q": [True, False]}
+    )
+    different_df.dtypes
+
+    dm = store_dataframes_as_dataset(
+        store_factory,
+        "multiple_partitioned_tables",
+        [{"data": {"table1": df, "table2": different_df}}],
+        partition_on="B",
+    )
+
+    sorted(dm.partitions.keys())
+
+
+As noted above, when data is appended to a dataset, ``kartothek`` guarantees it has
+the proper schema and partitioning.
+
+.. note:: Every partition must have data for every table. An empty dataframe in this
+          context is also considered as data.
+
+
+Secondary Indices
+-----------------
+
+The ability to build and maintain 'secondary' indices are an additional ability
+provided by ``kartothek``. In general, an index is a data structure used to improve
+the speed of read queries. In the context of ``kartothek`` an index is a data structure
+that contains a mapping of every unique value of a given column to references to all the
+partitions where this value occurs.
+
+Explicitly partitioning on a column internally creates a :class:`~kartothek.core.index.PartitionIndex`.
+While this index has a one-to-one mapping of column values to partition references,
+secondary indices have the advantage of being able to contain one-to-many mappings of
+column values to partition references.
+
+.. note::
+
+    The examples we've looked at so far have mostly used functions from the ``eager``
```

Comment:
I believe we covered this already in the getting started guide when the different backends are introduced, didn't we?